ObjectCreation/AeroFoil.FCMacro

```diff
@@ -0,0 +1,1747 @@
+            try:
+                App.ActiveDocument.removeObject(self.profileName)
+            except Exception:
+                pass
+            App.activeDocument().recompute()
+            return False
+        App.activeDocument().recompute()
+        FreeCADGui.activeDocument().activeView().viewFront()
+        FreeCADGui.SendMsgToActiveView("ViewFit")
+        return True
+
+
+
+# Sub-main Function Class
+# ------------------------------------------------------------------------------------------------
+
+class AeroFoilDialog:
+    def __init__(self):
+        # Variables Pertaining to the Entire Program
+        (
+            self.canMirror,
+            self.functionElements,
+            self.functionElements_Pos,
+            self.pointsX,
+            self.pointsY,
+        ) = (True, [], [], [], [])
+        # Variables Pertaining to AeroFoil_Dialog_1
+        self.airfoilType = 1
+        # Variables Pertaining to AeroFoil_Dialog_2a
+        self.airfoil4DNumber = ""
+        # Variables Pertaining to AeroFoil_Dialog_2b
+        self.airfoil5DNumber = ""
+        # Variables Pertaining to AeroFoil_Dialog_2c
+        self.airfoilProfileType, self.topCurveFunction, self.bottomCurveFunction = (
+            1,
+            "",
+            "",
+        )
+        # Variables Pertaining to AeroFoil_Dialog_2d
+        self.importFrom, self.mirroring = 1, False
+        # Variables Pertaining to AeroFoil_Dialog_2d1a
+        self.lineStart, self.lineEnd, self.decimalType = 0, 0, 1
+        # Variables Pertaining to AeroFoil_Dialog_2d1b
+        self.col1, self.col2, self.row1, self.row2 = 1, 2, 0, 0
+        # Variables Pertaining to AeroFoil_Dialog_2cd2
+        (
+            self.loadSuccess,
+            self.loadedFilePath,
+            self.loadedFile,
+            self.loadedFileContents,
+        ) = (False, [""], "", [])
+        # Variables Pertaining to AeroFoil_Dialog_3
+        (
+            self.refine,
+            self.refineParam,
+            self.multi,
+            self.multiParam,
+            self.closed_,
+            self.workbench,
+            self.curveType,
+            self.maxRefineParamWarned,
+            self.maxQuantityParamWarned,
+            self.chordLength,
+            self.chordUnits,
+            self.progressBar_,
+        ) = (False, 2, False, 2, False, 1, 1, False, False, 0, 1, 0)
+        # Actual Functions begin here
+        dialogIndex[0] = 1
+        self.createDialogs()
+
+    def d_(self, qcode, objStr):
+        qkey = [
+            0,
+            QPushButton,
+            QLineEdit,
+            QComboBox,
+            QRadioButton,
+            QCheckBox,
+            QSpinBox,
+            QDoubleSpinBox,
+            QLabel,
+            QProgressBar,
+        ]
+        return self.dialog.findChild(qkey[qcode], objStr)
+
+    def createDialogs(self):
+        global MD_
+        global dialogIndex
+        if dialogIndex[0] == 1:
```

Comment:
Maintainability would be improved by writing a module and a class for each dialog.

Reply:
That was the original thought. But that would take up lots of unnecessary lines of code due to repetition.
Hence I came up with the idea to treat the dialogs as a list of variables and jump back and forth by manipulating the list instead. 
I've added relevant docstrings in the macro explaining the same.

Reply:
Do you want me to try to solve this?

Reply:
Sure, Mr. Galou; I don't mind.